### ISSUE TITLE
Typescript-xml polymode

### ISSUE THREAD

**Issue by abeforgit** [polymode/polymode#320]
It seems the emacs-default `nxml-mode` doesn't play nice with polymode, at least not with [typescript-mode](https://github.com/emacs-typescript/typescript.el)

My setup:
```elisp
(define-hostmode poly-ts-hostmode
  :mode 'typescript-mode
  )
(define-innermode poly-xml-ts-innermode
  :mode 'xml-mode
  :head-matcher (rx "vdom`")
  :tail-matcher (rx "`")
  :head-mode 'host
  :tail-mode 'host
  )
(define-polymode poly-ts-xml-mode
  :hostmode 'poly-ts-hostmode
  :innermodes '(poly-xml-ts-innermode)
  )
(add-to-list 'auto-mode-alist '("\\.ts" . poly-ts-xml-mode))
```

(the head-matcher may seem weird but makes sense in my project)

The syntax highlighting breaks completely. It considers the whole buffer to be xml, even though mode-switching does work for other functionality. 

The setup works much better with `sgml-mode`, although sometimes highlighting only works for the first xml block, and the following errors appear in my modeline when I edit an xml block:
```elisp
(jit-lock--run-functions 1172 1281) [span 1172 1281 insert-text-command-test.ts[sgml]] -> (font-lock-default-fontify-region 1172 1281): Args out of range: 1171, 1171
``` 
(insert-text-command-test.ts is the file I'm editing on this particular occasion)
They don't seem to break anything significant though.

So in summary: default `xml-mode` breaks syntax highlighting (but seems to work otherwise) and the old `sgml-mode` works, but with some jankiness.
